src/de/mod.rs

```diff
@@ -461,34 +479,110 @@ impl<'a, 'de> de::Deserializer<'de> for &'a mut Deserializer<'de> {
+                                    Some(b'b') => 0x8,
+                                    Some(b'f') => 0xC,
+                                    Some(b'n') => b'\n',
+                                    Some(b'r') => b'\r',
+                                    Some(b't') => b'\t',
+                                    Some(b'u') => {
+                                        let (escape_sequence, remaining_escaped_string_bytes) =
+                                            escaped_string_bytes
+                                                .as_slice()
+                                                .split_first_chunk::<4>()
+                                                .ok_or(Error::InvalidEscapeSequence)?;
+
+                                        escaped_string_bytes =
+                                            remaining_escaped_string_bytes.iter();
+
+                                        let unescaped_char = core::str::from_utf8(escape_sequence)
+                                            .ok()
+                                            .and_then(|escape_sequence| {
+                                                u32::from_str_radix(escape_sequence, 16).ok()
+                                            })
+                                            .and_then(char::from_u32)
+                                            .ok_or(Error::InvalidEscapeSequence)?;
+
+                                        for &unescaped_byte in
+                                            unescaped_char.encode_utf8(&mut [0; 4]).as_bytes()
+                                        {
+                                            *string_unescape_buffer_slots
+                                                .next()
+                                                .ok_or(Error::EscapedStringIsTooLong)? =
+                                                unescaped_byte;
+                                        }
+
+                                        continue;
+                                    }
+                                    _ => return Err(Error::InvalidEscapeSequence),
+                                };
+
+                                *string_unescape_buffer_slots
+                                    .next()
+                                    .ok_or(Error::EscapedStringIsTooLong)? = unescaped_byte;
+                            }
+                            Some(c) => {
+                                *string_unescape_buffer_slots
+                                    .next()
+                                    .ok_or(Error::EscapedStringIsTooLong)? = c;
+                            }
+                        }
+                    }
+
+                    let remaining_length = string_unescape_buffer_slots.len();
+                    let unescaped_string_length =
+                        self.string_unescape_buffer.len() - remaining_length;
+
+                    visitor.visit_str(
```

Comment:
I'm a bit confused as to how this should behave if there's multiple entries within the JSON that are being de-escaped. Do we have a test that exercises a JSON with two fields that use escaped strings?

Reply:
The lifetime of string passed to `Visitor::visit_str` is ephemeral, and thus if two fields use escaped strings, the entire buffer is used for both de-escapings. It does mean that escaped strings cannot borrow from the input or the buffer, but I think that's OK.

Reply:
Yes, but does that cause issues in the actual deserialized output, since technically the unescaped string exists in the unescape buffer. If there's two unescaped strings, they'd have to live in that buffer, and that seems like there's a collision of memory there. Do we have a test to verify that works properly?

Reply:
> Ah does this work because the escaped string is moved after the de-escaping process? (i.e. into a heapless::String)

Yes